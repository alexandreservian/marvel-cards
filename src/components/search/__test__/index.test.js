import React from 'react';
import { render, fireEvent, cleanup } from 'test-utils';
import Search from '../index';

afterEach(cleanup);

describe('Search Component', () => {
  test('renders', () => {
    const { asFragment } = render(<Search />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('define prop defaultValue', () => {
    const { getByTestId } = render(<Search defaultValue="Wolverine" />);
    expect(getByTestId('input')).toHaveValue('Wolverine');
  });

  test('updates on change', () => {
    const { getByTestId } = render(<Search />);
    const input = getByTestId('input');
    fireEvent.change(input, { target: { value: 'spider-man' } });
    expect(input).toHaveValue('spider-man');
  });

  test('trigger onSearch', () => {
    const onSearch = jest.fn(value => value);
    const { getByTestId } = render(<Search onSearch={onSearch} />);
    const input = getByTestId('input');
    fireEvent.change(input, { target: { value: 'magneto' } });
    expect(onSearch).toHaveBeenCalled();
    expect(onSearch.mock.results[0].value).toBe('magneto');
  });
});
