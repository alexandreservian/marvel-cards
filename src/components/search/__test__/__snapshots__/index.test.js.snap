// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component renders 1`] = `
<DocumentFragment>
  <div
    class="css-1qx7s6a"
  >
    <div
      class="css-1db4zkn"
      width="95"
    >
      <img
        alt="logo"
        src="marvel-logo.svg"
      />
    </div>
    <div
      class="css-ftmyaz"
    >
      <i
        class="icon-search icon"
      />
      <input
        class="input"
        data-testid="input"
        placeholder="Find your favorite Marvel Hero!"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
